### PR TITLE
Update README.md with buffering info

### PR DESCRIPTION
**Pull request by raahi-chada2004**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -91,4 +93,12 @@ The plugin is designed to log and handle both client-side and server-side errors
...
+
+- Memory Buffering: Fluent Bit stores data chunks in memory by default. This method is fast but can lead to high memory usage under heavy load or network delays. To manage this, you can set a `Mem_Buf_Limit` field in the input section of the configuration field, which will restrict the memroy used by an input plugin, pausing data ingestion when the limit is reached. The tail input plugin is often sufficient with memory buffering due to its ability to track log offsets which minimizes data loss during pauses.
+- Filesystem Buffering: For greater data safety, filesystem buffering can be used. This requires finding a relevant input plugin that configures the storage.type to filesystem in its settings. This method stores data chunks both in memory and on disk, which gives control over memory usage.
```
Comment (yifatgortler) [inline README.md]: nit: `storage.type`

File: README.md [hunk 2]
```diff
@@ -91,4 +93,12 @@ The plugin is designed to log and handle both client-side and server-side errors
...
+
+More information on Fluent Bit buffering can be found here: [Fluent Bit: Official Manual - Buffering & Storage](https://docs.fluentbit.io/manual/administration/buffering-and-storage)
+
 For more information, look to [Fluentbit Official Guide to a Config File](https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/classic-mode/configuration-file)
```
Comment (yifatgortler) [inline README.md:130]: Maybe put this line under it's own title? To distinguish from the "Backpressure and Buffering" section?

File: README.md [hunk 3]
```diff
@@ -91,4 +93,12 @@ The plugin is designed to log and handle both client-side and server-side errors
 
 Regardless of the type of error, the plugin is built to maintain the flow of incoming data, ensuring that operations continue smoothly and that any issues are documented for troubleshooting.
 
+## Backpressure and Buffering
+This plugin utilizes dynamic stream scaling up when the rate of data being sent from Fluent Bit is too great for a single managed stream. However, to manage backpressure from the input/source to Fluent Bit itself, Fluent Bit implements its own buffering system where processed data is temporarily stored before being sent out. Fluent Bit primarily uses memory for buffering but can also utilize filesystem-based buffering for enhanced data safety.
+
+- Memory Buffering: Fluent Bit stores data chunks in memory by default. This method is fast but can lead to high memory usage under heavy load or network delays. To manage this, you can set a `Mem_Buf_Limit` field in the input section of the configuration field, which will restrict the memroy used by an input plugin, pausing data ingestion when the limit is reached. The tail input plugin is often sufficient with memory buffering due to its ability to track log offsets which minimizes data loss during pauses.
```
Comment (yifatgortler) [inline README.md]: Out of curiosity: Is the tail input plugin used for cases other than testing?

nit: "For example, the tail input plugin..." (or something similar)

Comment (raahi-chada2004) [reply README.md]: No it's not but with most log file input the tail input plugin would be utilized - so I was imagining our customers would likely use the tail input plugin.

File: README.md (hunk 1 above)
Comment (raahi-chada2004) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (raahi-chada2004) [reply README.md:130]: Done

File: README.md (hunk 3 above)
Comment (raahi-chada2004) [reply README.md]: Done

**Review by yifatgortler — APPROVED** [review]

**Pull request merged**